python/tchannel/sync.py

```diff
@@ -0,0 +1,140 @@
+# Copyright (c) 2015 Uber Technologies, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+from __future__ import absolute_import
+
+from collections import namedtuple
+
+from threadloop import ThreadLoop
+from tornado import gen
+
+from . import glossary
+from . import tornado as async
+
+
+class TChannelSyncClient(object):
+    """Make synchronous TChannel requests.
+
+    This client does not support incoming connections or requests- this is
+    a uni-directional client only.
+
+    The client is implemented on top of the Tornado-based implementation and
+    starts and stops IOLoops on-demand.
+
+    .. code-block:: python
+
+        client = TChannelSyncClient()
+        response = client.request(
+            hostport='localhost:4040',
+            service='HelloService',
+        ).send(
+            'hello', None, json.dumps({"name": "World"})
+        )
+
+    """
+
+    def __init__(self, name, process_name=None, known_peers=None, trace=False):
+        """Initialize a new TChannelClient.
+
+        :param process_name:
+            Name of the calling process. Used for logging purposes only.
```

Comment:
I wouldn't downplay the importance of the name, it's used for logging, tracing, stats, etc.